app/models/manageiq/providers/openstack/cloud_manager/provision_workflow.rb

```diff
@@ -1,8 +1,10 @@
 class ManageIQ::Providers::Openstack::CloudManager::ProvisionWorkflow < ::MiqProvisionCloudWorkflow
   def allowed_instance_types(_options = {})
-    source = load_ar_obj(get_source_vm)
-    ems = get_targets_for_ems(source, :cloud_filter, Flavor, 'flavors')
-    ems.each_with_object({}) { |f, h| h[f.id] = display_name_for_name_description(f) }
+    source                = load_ar_obj(get_source_vm)
+    flavors               = get_targets_for_ems(source, :cloud_filter, Flavor, 'flavors')
+    minimum_disk_required = source.hardware.size_on_disk
```

Comment:
I think it should take max(size_on_disk, min_disk) where min_disk is an optional attribute from the API  https://github.com/fog/fog/blob/master/lib/fog/openstack/models/image_v2/image.rb#L25

I think we are not storing min_disk now, we should.

The same will apply for the min_ram attribute. https://github.com/fog/fog/blob/master/lib/fog/openstack/models/image_v2/image.rb#L30